app/Http/Controllers/admin/TopicController.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+namespace App\Http\Controllers\admin;
+
+use App\Http\Requests\TopicFormRequest;
+use App\Models\Topic;
+use App\Repositories\Contracts\TopicRepository;
+use Carbon\Carbon;
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\File;
+use Illuminate\Support\Facades\Input;
+use Illuminate\Support\Facades\Redirect;
+use Illuminate\Support\Facades\Storage;
+
+class TopicController extends Controller
+{
+
+    protected $topic;
+
+    public function __construct(TopicRepository $topic)
+    {
+        $this->topic = $topic;
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $topics = $this->topic->getAll();
+        $test_model = new Topic();
+        $fillable_columns = $test_model->getFillable();
+        foreach ($fillable_columns as $key => $value) {
+            $test_columns[$value] = $value;
+        }
+
+        return view('admin.topic.index', compact('topics', 'test_columns'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        return view('admin.topic.create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(TopicFormRequest $request)
+    {
+        if (Auth::check()) {
```

Comment:
**Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 64</br>Variable "user_id" is not in valid camel caps format